internal/handlers/common/filter.go

```diff
@@ -58,11 +59,33 @@ func filterDocumentPair(doc *types.Document, filterKey string, filterValue any)
 		if err != nil {
 			return false, nil // no error - the field is just not present
 		}
-		var ok bool
-		if doc, ok = docValue.(*types.Document); !ok {
-			return false, nil // no error - the field is just not present
+
+		switch docValue := docValue.(type) {
+		case *types.Document:
+			doc = docValue
+			filterKey = path.Suffix()
+		case *types.Array:
+			index, err := strconv.Atoi(path.Suffix())
+			if err != nil {
+				return false, nil
+			}
+
+			if docValue.Len() == 0 {
+				return false, nil
+			}
+
+			value, err := docValue.Get(index)
+			if err != nil {
+				return false, err
+			}
+
+			// We shouldn't compare with embedded array values.
```

Comment:
A comment is not clear to me. Why do we create an empty array here? Can you please add a code snippet to note? Thank you.

Reply:
Fixed it, we should just return `false`.

Reply:
+1! Me neither! I don't understand why we shouldn't compare!

Reply:
What should the right comment be?

Reply:
> What should the right comment be?

`MongoDb compatibility`: <that comment> 

Can it be added?

Reply:
I think that comment is a bit redundant in FerretDB: the whole project is about MongoDB compatibility. Being compatible is the norm; being incompatible is an exception that should be documented.

Reply:
Removed.